core/identity-management/iin-agent/src/server.ts

```diff
@@ -130,9 +155,32 @@ iinAgentServer.addService(iin_agent_pb_grpc.IINAgentService, {
 
 // Bootstrapping
 const configSetup = async () => {
+    // Initiate agent's ledger
+    const securityDomain = process.env.SECURITY_DOMAIN ? process.env.SECURITY_DOMAIN : 'network1';
+    const memberId = process.env.MEMBER_ID ? process.env.MEMBER_ID : 'Org1MSP';
+    const ledgerBase = getLedgerBase(securityDomain, memberId);
+    await ledgerBase.init();
+    console.log("Setup compelete.");
+    
     // TODO
 };
 
+const loopSyncExternalState = async () => {
+    const delayTime: number = parseInt(process.env.SYNC_PERIOD ? process.env.SYNC_PERIOD : '3600');
+    const securityDomain = process.env.SECURITY_DOMAIN ? process.env.SECURITY_DOMAIN : 'network1';
+    const memberId = process.env.MEMBER_ID ? process.env.MEMBER_ID : 'Org1MSP';
+    const flagSync = process.env.AUTO_SYNC === 'false' ? false : true
+    while (flagSync) {
+        const secDomDNS = getAllSecurityDomainDNS();
+        for (const securityDomain in secDomDNS) {
+            const foreignSecurityDomain = new iin_agent_pb.SecurityDomainMemberIdentity();
+            foreignSecurityDomain.setSecurityDomain(securityDomain);
+            syncExternalStateFromIINAgent(foreignSecurityDomain, securityDomain, memberId);
```

Comment:
Given how our protocol works, this doesn't need to be in a loop. It can be called exactly once, on one of the selected local IIN Agents. That agent can then orchestrate collection of signatures from other agents, which in turn will sync from external agents after seeing that no memberships are stored in memory.

Reply:
I didn't exactly get your point. This is in loop, to sync memberships from all foreign networks defined in `dnsconfig.json`.

Reply:
Ah....sorry, I misread what you were looping on. I thought you were iterating over the orgs.